### PR TITLE
Pricing plans feature

### PR DESCRIPTION
**Pull request by PinkCode15**

### CODE REVIEW

File: app/Models/User.php [hunk 1]
```diff
@@ -138,4 +135,95 @@ public static function exists(string $email): bool
...
+    /**
+     * Checks if subscription limit for a user based on their pricing plan is exceeded
+     * 
+     * @return bool
+     */
+    public function isSubscriptionLimitReached(): bool
+    {
+        if ($this->pricingPlan) {
+            
+            $subscriptionCount = $this->subscriptions()->where('status', 'active')->count();
```
Comment (IDTitanium) [inline app/Models/User.php:207]: Do expired subscriptions count? Does this status field get updated for expired subs?

Comment (PinkCode15) [reply app/Models/User.php:207]: No, it only counts the one we are tracking which are the active subscriptions. 
Yes the status field gets updated when a subscription expires.

File: database/seeders/data/pricing_plans.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        "amount":1500,
+        "period": "monthly"
+    },
+    {
+        "name":"Premium",
+        "details":["Track unlimited subscriptions at a time", "View detailed report about subscriptions", "Access to chrome extension, mobile application, and web application"],
+        "subscription_limit": null,
+        "amount":15600,
+        "period": "yearly"
+    }
```
Comment (IDTitanium) [inline database/seeders/data/pricing_plans.json:36]: How will the frontend be able to condense the two premiums into one view?
There should be only one object per plan, so it's easier to use.

Comment (PinkCode15) [reply database/seeders/data/pricing_plans.json:36]: Well, I can just format the response object to have it condensed into 1 object for each plan

**Review by IDTitanium — APPROVED** [review]

**Pull request merged**